projects/web3/eth_wallet/ta/src/main.rs

```diff
@@ -18,37 +18,53 @@
 #![no_main]
 
 mod hash;
-mod secure_storage;
 mod wallet;
 
-use crate::secure_storage::{
-    delete_from_secure_storage, load_from_secure_storage, save_in_secure_storage,
-};
 use optee_utee::{
     ta_close_session, ta_create, ta_destroy, ta_invoke_command, ta_open_session, trace_println,
 };
 use optee_utee::{Error, ErrorKind, Parameters};
 use proto::Command;
+use secure_db::SecureStorageClient;
 
 use anyhow::{anyhow, bail, Result};
-use std::convert::TryInto;
 use std::io::Write;
 use wallet::Wallet;
 
+const DB_NAME: &str = "eth_wallet_db";
+
+// Define session context structure
+pub struct WalletSession {
+    db_client: SecureStorageClient,
+}
+
+impl Default for WalletSession {
+    fn default() -> Self {
+        Self {
+            db_client: SecureStorageClient::open(DB_NAME)
+                .expect("Failed to create SecureStorageClient"),
+        }
+    }
+}
+
 #[ta_create]
 fn create() -> optee_utee::Result<()> {
     trace_println!("[+] TA create");
     Ok(())
 }
 
 #[ta_open_session]
-fn open_session(_params: &mut Parameters) -> optee_utee::Result<()> {
+fn open_session(_params: &mut Parameters, _sess_ctx: &mut WalletSession) -> optee_utee::Result<()> {
```

Comment:
What is the purpose of this unused parameter?

Reply:
Add explanation as code comment:
// The _sess_ctx: &mut WalletSession is explicitly defined here to fit into the
// macros definition (optee-utee/macros/src/lib.rs), meaning that we have a
// context for the session which would be initialized in the open_session() method.
// The context is initialized automatically by calling WalletSession::default()
// in the macro.

Reply:
Given that the pattern also exists in other examples: https://github.com/apache/incubator-teaclave-trustzone-sdk/blob/main/examples/acipher-rs/ta/src/main.rs#L32-L54, also in `hotp-rs` `aes-rs` `digest-rs` and so on, I think it's reasonable to use the same pattern here for now, we can open another PR to revise the session context handling then update all related examples.
If it's not acceptable, I'm okay to rollback the `eth_wallet` example to the version doesn't use the secure_db crate.

@m4sterchain Which approach would you prefer?

Reply:
No problem. We can proceed and resolve `sessionContext` together in future PRs.